thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/diffsummary/DimNameValueCostEntry.java

```diff
@@ -1,20 +1,27 @@
 package com.linkedin.thirdeye.client.diffsummary;
 
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
 public class DimNameValueCostEntry implements Comparable<DimNameValueCostEntry>{
-  private double contributionFactor;
   private String dimName;
   private String dimValue;
   private double cost;
+  private double contributionFactor;
   private double currentValue;
   private double baselineValue;
 
-  public DimNameValueCostEntry(String dimension, String dimValue, double dimValueCost, double contributionFactor, double currentValue, double baselineValue) {
-    this.dimName = dimension;
-    this.dimValue = dimValue;
-    this.cost = dimValueCost;
-    this.contributionFactor = contributionFactor;
-    this.currentValue = currentValue;
+  public DimNameValueCostEntry(String dimensionName, String dimensionValue, double baselineValue, double currentValue,
```

Comment:
why not pass the globalBaseline and globalCurrent, you can have methods such as getContributionFactor etc

Reply:
There are two reasons that I didn't pass globalBaseline and globalCurrent to this class:
1. This class is used to store the information to be presented at the frontend, so it shouldn't contain any computational logic. The contribution factor should be calculated and pass to the instance of this class.
2. The value of  globalBaseline and globalCurrent are globally available in the summary that is pass to the front end. So the global values that are stored in this class will not be used in anyway.